src/App.tsx

```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  currentGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    currentGoods: ['Jam'],
+  };
+
+  clear = () => {
+    this.setState({ currentGoods: [] });
+  };
+
+  checkGoods = (includeGood: string) => {
+    return this.state.currentGoods.includes(includeGood);
+  };
+
+  selectAndRemove = (good: string) => {
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    const newGood = [...this.state.currentGoods];
+
+    if (newGood.includes(good)) {
+      newGood.splice(newGood.indexOf(good), 1);
+    } else {
+      newGood.push(good);
+    }
+
+    return this.setState({ currentGoods: newGood });
+  };
+
+  printGoods = () => {
+    const newGood = [...this.state.currentGoods];
+    const lastElem = newGood.slice(-1, newGood.length);
+
+    if (newGood.length > 1) {
+      return `${newGood.slice(0, newGood.length - 1).join(', ')} and ${lastElem} are selected.`;
+    }
+
+    return `${newGood} is selected.`;
+  };
+
+  render() {
+    const { currentGoods } = this.state;
+
+    return (
+      <div className="good content is-medium">
+        <button
+          type="button"
+          onClick={this.clear}
+          className={classNames('goood__clear button is-danger', {
+            'is-invisible': currentGoods.length === 0,
+          })}
+        >
+          Clear
+        </button>
+        <h1 className="good__title">Selected good: </h1>
+        <h2 className="good__subtitle">
+          {currentGoods.length === 0 ? (
+            <>
+              No goods selected
+            </>
+          ) : <>{this.printGoods()}</>}
+        </h2>
+        <ul className="good__list">
+          {
+            goodsFromServer.map(good => (
+              <>
+                <li key={good} className="good__item">
+                  {good}
+                  <button
+                    type="button"
+                    onClick={() => {
+                      this.selectAndRemove(good);
+                    }}
+                    className={classNames(
+                      this.checkGoods(good) ? (
+                        'button is-danger is-rounded'
+                      ) : 'button is-primary is-rounded',
```

Comment:
DRY use `"button is-rounded"` as default classes here, and `is-danger` or `is-primary`
optionaly you can create a variable inside `.map` method
`const checked = this.checkGoods(good);`
finally you will have:
```
className={classNames(
                     'button is-rounded'
                      {
                        'is-danger': checked,
                        'is-primary': !checked,
                      },
```